.github/CONTRIBUTING.md

```diff
@@ -1,12 +1,31 @@
 # How to use Jedis Github Issue
 
-* Github issues SHOULD ONLY BE USED to report bugs, and for DETAILED feature requests. Everything else belongs to the Jedis Google Group.
+* Github issues SHOULD ONLY BE USED to report bugs and for DETAILED feature requests. Everything else belongs to the [Jedis Google Group](https://groups.google.com/g/jedis_redis) or [Jedis GitHub Discussions](https://github.com/redis/jedis/discussions).
 
-Jedis Google Group address:
-  
-https://groups.google.com/forum/?fromgroups#!forum/jedis_redis
+Please post General questions to Google Group or Github discussions. It can be closed without answer when posted to Github issues.
 
-Please post General questions to Google Group. It can be closed without answer when posted to Github issue.
+# How to contribute by Pull Request
+
+1. Fork Jedis on github (https://help.github.com/articles/fork-a-repo/)
+2. Create a topic branch (git checkout -b my_branch)
+3. Push to your branch (git push origin my_branch)
+4. Post a pull request on github (https://help.github.com/articles/creating-a-pull-request/)
+
+It is recommended to create branch with meaningful name, not modifying master branch directly.
+
+Please add unit tests in order to prove your modification works smoothly. And please make sure your modification passes all unit tests.
+
+# Jedis Test Environment
+
+Jedis unit tests run with latest [Redis unstable branch](https://github.com/redis/redis/tree/unstable).
+Please let them prepared and installed.
+
+Jedis unit tests use many Redis instances, so we use ```Makefile``` to prepare environment. 
+
+You can start test with ```make test```.
+You can set up test environments by ```make start```, and tear down environments by ```make stop```.
+
+If one or some of unit tests in current master branch of Jedis fails with Redis unstable branch, please post it to Github issue, and go ahead with other unit tests at your work.
```

Comment:
```
If unit tests fail in the master branch of Jedis fails, while running against the Redis unstable branch, please create a Github issue, and continue testing.
```

Reply:
I have removed the sentence. Our flaky tests could cause a current of issues.